python/ray/data/datasource/file_based_datasource.py

```diff
@@ -61,6 +62,19 @@
 OPEN_FILE_MAX_ATTEMPTS = 10
 
 
+@DeveloperAPI
+@dataclass
+class FileShuffleConfig:
+    """Configuration for file shuffling."""
```

Comment:
Can you add a docstring example and arg description? Maybe something like:

```
# (Pseudocode. You'll need to make this actually run or CI will fail)
>>> import ray
>>> import FileShuffleConfig
>>> read_parquet(...., shuffle=FileShuffleConfig(seed=42))
```

Also, maybe add a `..note` that even if you seed the file shuffling, you might still get a somewhat random row order since tasks complete in non-deterministic order.